src/main/java/org/icatproject/topcat/web/rest/UserResource.java

```diff
@@ -62,6 +63,20 @@
 	@PersistenceContext(unitName = "topcat")
 	EntityManager em;
 
+	/**
+	 * Returns the cart userName, which is either the ICAT userName if the user isn't anonUserName,
+	 * or it's the ICAT userName plus the sessionId if it is the anon user name
+	 */
+	private String getCartUserName(String userName, String sessionId) {
+		Properties properties = Properties.getInstance();
+		String anonUserName = properties.getProperty("anonUserName", "");
```

Comment:
Couldn't the properties file access be added to a constructor? It seems that the properties should only be accessed once at instantiation not every time this method is called. This is how the [IdsClient](https://github.com/icatproject/topcat/blob/656e8260216bf1048260ae4ff51f6acfb8af5588/src/main/java/org/icatproject/topcat/IdsClient.java#L38) does it.

Reply:
True, that would make more sense